### PR TITLE
use the same task runner for platform&render tasks

### LINKED ISSUE #133: Segmentation fault: 11 when resizing window on MacOS
Initially screen is empty, then when I resize it - application crashes with message in issue title.
After maximizing application works fine.

### PR DESCRIPTION
**Pull request by pchampio**
This fixes the issue where on MacOS the embedder crashes during window resizing.
fixes #133

This doesn't fixes the jankyness during resize.
Follow https://github.com/flutter/flutter/issues/44136 for more info.

Need to wait for https://github.com/flutter/engine/pull/13124 to be rolled to the flutter beta branch.

### CODE REVIEW

File: embedder/embedder_helper.c [hunk 1]
```diff
@@ -45,20 +47,24 @@ FlutterEngineResult runFlutter(void *user_data, FlutterEngine *engine,
...
   platform_task_runner.runs_task_on_current_thread_callback =
       proxy_runs_task_on_current_thread_callback;
   platform_task_runner.post_task_callback = proxy_post_task_callback;
+  platform_task_runner.identifier = ++gTaskRunnerIdentifiers;
 
   FlutterCustomTaskRunners custom_task_runners = {};
   custom_task_runners.struct_size = sizeof(FlutterCustomTaskRunners);
+  // Render task and platform task are handled by the same TaskRunner
   custom_task_runners.platform_task_runner = &platform_task_runner;
+  custom_task_runners.render_task_runner = &platform_task_runner;
```
Comment (pchampio) [inline embedder/embedder_helper.c]: This is the important line!

**Comment by pchampio** [comment]
Beta has been updated.
This can be merged.
Test with: `hover run -b '@no_resize_crash'`

File: application.go [hunk 2]
```diff
@@ -233,14 +233,32 @@ func (a *Application) Run() error {
...
+		}
+		os.Exit(1)
+	}
+
 	// Start the engine
-	result := a.engine.Run(unsafe.Pointer(&flutterEnginePointer), a.config.vmArguments)
+	result = a.engine.Run()
 	if result != embedder.ResultSuccess {
 		switch result {
 		case embedder.ResultInvalidLibraryVersion:
```
Comment (GeertJohan) [inline application.go]: Can this statuscode be returned by both Init AND Run?? You'd expect that if Init is fine with the lib version, then Run will be as well?

File: application.go [hunk 3]
```diff
@@ -233,14 +233,32 @@ func (a *Application) Run() error {
...
 			fmt.Printf("go-flutter: engine.Run() returned result code %d (invalid library version)\n", result)
 		case embedder.ResultInvalidArguments:
```
Comment (GeertJohan) [inline application.go]: Can this still be returned by Run()? There are no arguments anymore...

**Review by GeertJohan — APPROVED** [review]
Two questions about the result codes.
I haven't tested this on my machine yet, but the code LGTM.

File: application.go (hunk 3 above)
Comment (pchampio) [reply application.go]: There is no documentation about which result can be returned or not.
And in the C call, there is an argument.

File: application.go (hunk 2 above)
Comment (pchampio) [reply application.go]: As there is no documentation about this on the abi: https://github.com/go-flutter-desktop/go-flutter/blob/50bccfd1b0b3839c25da4c7c95095d0cebc2db32/embedder/embedder.h#L1040-L1065
I've left all cases.

**Pull request merged**